src/main/kotlin/app/revanced/patches/youtube/misc/settings/bytecode/patch/SettingsPatch.kt

```diff
@@ -39,13 +42,31 @@ import java.io.Closeable
 @SettingsCompatibility
 @Version("0.0.1")
 class SettingsPatch : BytecodePatch(
-    listOf(LicenseActivityFingerprint, ReVancedSettingsActivityFingerprint, ThemeSetterFingerprint)
+    listOf(LicenseActivityFingerprint, ReVancedSettingsActivityFingerprint, ThemeSetterFingerprint, ThemeSetterParentFingerprint)
 ) {
     override fun execute(context: BytecodeContext): PatchResult {
         val licenseActivityResult = LicenseActivityFingerprint.result!!
         val settingsResult = ReVancedSettingsActivityFingerprint.result!!
         val themeSetterResult = ThemeSetterFingerprint.result!!
 
+        val themeSetterParentResult = ThemeSetterParentFingerprint.result!!
+        val themeSetterParentMethod = themeSetterParentResult.mutableMethod
+        val themeSetterParentInstructions = themeSetterParentMethod.implementation!!.instructions
+        val themeSetterParentResultIndex = themeSetterParentResult.scanResult.patternScanResult!!.endIndex
+
+        themeSetterParentMethod.addInstructions(
```

Comment:
Why is the addition call here required, below setTheme is already called.

Reply:
Because if you change the color directly from the app, this patch doesn't work. For this reason, an extra themeSet() is needed.

Reply:
Try it youself to better understand what I mean.